### PR TITLE
RA-500: Make sure similar name searching during registration works out o...

### PR DESCRIPTION
**Pull request by pmuchowski**
...f the box

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/registrationcore/api/search/NamePhoneticsPatientSearchAlgorithm.java [hunk 1]
```diff
@@ -87,4 +88,18 @@ protected Criteria addNameCriteria(Criteria criteria, Patient patient) {
...
 	}
+
+	/**
+	 * Returns a score higher than 0 if value and matches strings are not empty.
+	 *
+	 * @param value
+	 * @param matches
+	 * @return 0 if value or matches strings are empty else return 1
+	 */
+	public double countStartWithScoreForField(String value, String matches) {
```
Comment (wluyima) [inline api/src/main/java/org/openmrs/module/registrationcore/api/search/NamePhoneticsPatientSearchAlgorithm.java]: So where is this called from?

Comment (pmuchowski) [inline api/src/main/java/org/openmrs/module/registrationcore/api/search/NamePhoneticsPatientSearchAlgorithm.java]: It's called in BasicSimilarPatientSearchAlgorithm.findSimilarPatients

Comment (wluyima) [inline api/src/main/java/org/openmrs/module/registrationcore/api/search/NamePhoneticsPatientSearchAlgorithm.java]: Are you saying one algorithm calls another algorithm? Is this true @djazayeri ?

Comment (pmuchowski) [inline api/src/main/java/org/openmrs/module/registrationcore/api/search/NamePhoneticsPatientSearchAlgorithm.java]: NamePhoneticsPatientSearchAlgorithm extends BasicSimilarPatientSearchAlgorithm

Comment (wluyima) [inline api/src/main/java/org/openmrs/module/registrationcore/api/search/NamePhoneticsPatientSearchAlgorithm.java]: Ok, then i think you need to put an @Override

**Pull request merged**